manifests/journal_upload.pp

```diff
@@ -0,0 +1,38 @@
+# @api private
+# @summary This class manages and configures journal-upload.
+# @see https://www.freedesktop.org/software/systemd/man/journald.conf.html
+#
+# @param package_name
+#  name of the package to install for the functionality
+#
+class systemd::journal_upload (
+  Optional[String[1]] $package_name = undef,
+) {
+  assert_private()
+
+  if $package_name {
+    stdlib::ensure_packages($package_name)
+  }
+
+  service { 'systemd-journal-upload':
+    ensure => running,
+  }
+  $systemd::journal_upload_settings.each |$option, $value| {
+    ini_setting {
+      $option:
```

Comment:
I think the following syntax is more common in our code base:
```suggestion
    ini_setting {  $option:
```

Reply:
took it from journald.pp, will change.